[Bug]: Ethereum Client failing with nil pointer error on nullchain

### Problem encountered

When running nullchain following a recent update the core crashes with a nil pointer exception

### Observed behaviour

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x13b2e07]

goroutine 181 [running]:
code.vegaprotocol.io/vega/core/client/eth.(*Client).BlockByNumber(0x0?, {0x3f58ca8?, 0xc000e9cc30?}, 0x0?)
	<autogenerated>:1 +0x27
code.vegaprotocol.io/vega/core/datasource/external/ethcall.(*Engine).Poll(0xc0002df2d0, {0x3f58ca8, 0xc000e9cc30}, {0x4927d0?, 0x4a817c800?, 0x5f615c0?})
	/jenkins/workspace/common/vega-market-sim/extern/vega/core/datasource/external/ethcall/engine.go:200 +0x74
code.vegaprotocol.io/vega/core/datasource/external/ethcall.(*Engine).Start.func1.1()
	/jenkins/workspace/common/vega-market-sim/extern/vega/core/datasource/external/ethcall/engine.go:89 +0x51
code.vegaprotocol.io/vega/core/datasource/external/ethcall.(*poller).Loop(0xc001851d40, 0xc0065edfa8)
	/jenkins/workspace/common/vega-market-sim/extern/vega/core/datasource/external/ethcall/engine.go:319 +0xa9
code.vegaprotocol.io/vega/core/datasource/external/ethcall.(*Engine).Start.func1()
	/jenkins/workspace/common/vega-market-sim/extern/vega/core/datasource/external/ethcall/engine.go:88 +0xf5
created by code.vegaprotocol.io/vega/core/datasource/external/ethcall.(*Engine).Start
	/jenkins/workspace/common/vega-market-sim/extern/vega/core/datasource/external/ethcall/engine.go:78 +0x65
```

### Expected behaviour

No error

### Steps to reproduce

```bash
Run fuzz test within Market Sim
```


### Software version

Latest develop

### Failing test

_No response_

### Jenkins run

_No response_

### Configuration used

_No response_

### Relevant log output

_No response_

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -109,6 +109,7 @@
 - [8739](https://github.com/vegaprotocol/vega/issues/8739) - Cancel orders for rejected markets.
 - [9118](https://github.com/vegaprotocol/vega/issues/9118) - Improve list stop orders error message
 - [9105](https://github.com/vegaprotocol/vega/issues/9105) - Truncate virtual stake decimal places
+- [9348](https://github.com/vegaprotocol/vega/issues/9348) - Nil pointer error in Ethereum call engine when running with null block chain 
 - [8800](https://github.com/vegaprotocol/vega/issues/8800) - `expiresAt` is always null in the Stop Orders `API`.
 - [8796](https://github.com/vegaprotocol/vega/issues/8796) - Avoid updating active proposals slice while iterating over it.
 - [8631](https://github.com/vegaprotocol/vega/issues/8631) - Prevent duplicate Ethereum call chain event after snapshot start

--- a/core/datasource/external/ethcall/engine.go
+++ b/core/datasource/external/ethcall/engine.go
@@ -4,6 +4,7 @@ import (
 	"context"
 	"fmt"
 	"math/big"
+	"reflect"
 	"sync"
 	"time"
 
@@ -74,7 +75,7 @@ func NewEngine(log *logging.Logger, cfg Config, isValidator bool, client EthRead
 // Start starts the polling of the Ethereum bridges, listens to the events
 // they emit and forward it to the network.
 func (e *Engine) Start() {
-	if e.isValidator && e.client != nil {
+	if e.isValidator && !reflect.ValueOf(e.client).IsNil() {
 		go func() {
 			ctx, cancelEthereumQueries := context.WithCancel(context.Background())
 			defer cancelEthereumQueries()

